### DISCUSSION TITLE
How to open an Excalidraw file from Ubuntu's file manager?

### DISCUSSION THREAD

**Discussion by sojusnik** [excalidraw/excalidraw#2714] [Q&A] [answered]
According to [this](https://web.dev/file-handling/#demo), it's possible to open Excalidraw files directly from the file manager, but how to accomplish this with Ubuntu's file manager (Gnome Files)? In doing so, the file should be opened by an PWA of Excalidraw, in my case created by [Brave](https://en.wikipedia.org/wiki/Brave_(web_browser)).

**Comment by dwelle** [comment, marked as answer]
What does the workflow of opening an `.excalidraw` file look like on Ubuntu?

**Reply by sojusnik** [reply]
Find a `.excalidraw` file in the file manager, double-click (or simple-click) on it to open, preferably in a PWA. It's the same workflow as with any other file, like `.pdf` etc.

Or do you mean something else?

**Reply by dwelle** [reply]
I meant, does it currently differ from what it looks like on Win/Mac? On windows, if the extension isn't associated yet, it'll ask you what program you want it to open with. You can then choose Chrome which will open the PWA.

**Reply by sojusnik** [reply]
The thing is that if I associate `.excalidraw` to open with Brave, then this file opens like this:

![grafik](https://user-images.githubusercontent.com/1245718/103589495-ca1b4d00-4eeb-11eb-9456-13b9ecc5f017.png)

**Reply by dwelle** [reply]
I forgot, you need to have `brave://flags/#file-handling-api` enabled. After that, Brave should add itself into apps you can open the `.excalidraw` file with (it should be a separate Brave "app", AFAIK you can't just open the file in the normal Brave app — that will end up with what you're seeing above).

**Reply by sojusnik** [reply]
Switching `brave://flags/#file-handling-api` to enabled did the trick, many thanks!
